### PR TITLE
Implement Rake task to check REFERENCE.md

### PR DESCRIPTION
**Pull request by bastelfreak**
I think it's not that beautiful to call a rake task from the module, but modulesync_config templates it and I cannot move the reference task to voxpupuli-release. I did  some local testing and the check seems to work fine.

### CODE REVIEW

File: lib/voxpupuli/release/rake_tasks.rb [hunk 1]
```diff
@@ -49,3 +50,17 @@
     fail "Unable to find a CHANGELOG.md entry for the #{v} release."
   end
 end
+
+desc 'Check REFERENCE.md.'
+task :check_reference_md do
+  if File.exists?('REFERENCE.md')
+    # the task is defined in the Rakefile in each puppet module
+    # https://github.com/voxpupuli/modulesync_config/blob/a7c683a828332429852a6521598c7ce8652f016a/moduleroot/Rakefile.erb#L40-L44
+    Rake::Task['reference'].invoke
```
Comment (ekohl) [inline lib/voxpupuli/release/rake_tasks.rb:59]: Perhaps it should check if there's a diff before regenerating and exit before that?

Comment (bastelfreak) [reply lib/voxpupuli/release/rake_tasks.rb:61]: good idea, added it

**Review by smortex — APPROVED** [review]

**Comment by ekohl** [comment]
So I thought a bit more about this. Isn't this something that we should implement in puppet-strings directly so it's usable for the wider community?

There's this definition of the reference generation:
https://github.com/puppetlabs/puppet-strings/blob/5955970a8c7bad66e7b20da344afa4d245f33ab3/lib/puppet-strings/tasks/generate.rb#L50-L52

Can we also modify one that and write it to a temporary file and diff the two as pure Ruby? That way you never overwrite what's on disk (making it non-destructive) and doesn't rely on git.

**Comment by bastelfreak** [comment]
I like the idea to make this more flexible (move it directly to puppet-strings), but I don't know how to implement this properly in puppet-strings. I thought about making this non-destructive by writing into another file, but I didn't find a proper option in puppet-strings for that. If someone here wants to help out let me know.

**Comment by ekohl** [comment]
Would https://github.com/puppetlabs/puppet-strings/pull/291 address that?

**Pull request closed without merging**